src/notation/internal/notationactioncontroller.cpp

```diff
@@ -2254,9 +2254,11 @@ bool NotationActionController::startNoteInputAllowed() const
 
 void NotationActionController::startNoteInput()
 {
-    INotationNoteInputPtr noteInput = currentNotationNoteInput();
-    if (noteInput) {
-        noteInput->startNoteInput(configuration()->defaultNoteInputMethod());
+    if (!configuration()->enterNoteInputModeOnKeyboardEntry()) {
```

Comment:
This seems not the right place for this check; it is likely to have undesirable effects this way. Instead, I would add it in more specific places, even if that means you have to do the same check in multiple methods.